azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/manager/AudioSessionManager.kt

```diff
@@ -35,7 +36,16 @@ internal class AudioSessionManager(
             false
         }
 
+    private val bluetoothDeviceName: String
+        get() = try {
+            bluetoothAudioProxy?.connectedDevices?.firstOrNull()?.name
+                ?: context.getString(R.string.azure_communication_ui_setup_audio_device_bluetooth)
+        } catch (exception: SecurityException) {
+            context.getString(R.string.azure_communication_ui_setup_audio_device_bluetooth)
+        }
+
     private var previousAudioDeviceSelectionStatus: AudioDeviceSelectionStatus? = null
+    private var priorToBluetoothAudioSelectionStatus: AudioDeviceSelectionStatus? = null
 
     suspend fun start() {
         BluetoothAdapter.getDefaultAdapter()?.run {
```

Comment:
getDefaultAdapter() is deprecated in API31
https://developer.android.com/reference/android/bluetooth/BluetoothAdapter#getDefaultAdapter()

Reply:
acutally, @ahammer can you create a separate refactor ticket for the deprecated function, and I will approve this one

Reply:
acutally, @ahammer can you create a separate refactor ticket for the deprecated function, and I will approve this one

Reply:
Hmmm OK, I'll take a look.